src/HPA-UMAP-3D.imjoy.html

```diff
@@ -132,7 +132,20 @@
                     await api.showStatus('done')
                 },
                 async export_csv(){
-                    console.log('Filtered data: ', self.filtered_table)
+                    var export_string = "ID,X,Y,Z,GENE,LOCATIONS,HPA_URL\n"
+                    for (var current_index in self.filtered_table) {
+                        var current_data = self.filtered_table[current_index]
+                        export_string += current_data["id"] + ","
+                        export_string += current_data["x"] + ","
+                        export_string += current_data["y"] + ","
+                        export_string += current_data["z"] + ","
+                        export_string += current_data["gene"] + ","
+                        export_string += "\"" + current_data["location"] + "\","
+                        export_string += current_data["x"] + "," 
+                        export_string += 'https://www.proteinatlas.org/'+current_data["gene_ensembl_id"]+'-'+current_data["gene"]+'/cell/'
+                        export_string += "\n"
+                    }
+                    api.exportFile(export_string, 'export.csv')
```

Comment:
As we discussed, perhaps better to use ; as separate for csv file.

Reply:
That is untrue.  With the filename csv you should definitely use commas. It is easy to use in Excel too, despite the commas.

Reply:
Here is a thread about it: https://stackoverflow.com/questions/10140999/csv-with-comma-or-semicolon

With also one solution for MS Excel: https://stackoverflow.com/questions/17953679/how-to-correctly-display-csv-files-within-excel-2013#19769403
Not sure how that behaves on other reader though.

Reply:
I would rather follow the RFC (https://www.ietf.org/rfc/rfc4180.txt) than follow Microsofts bullshit.